compiler/src/Language/Bond/Codegen/Cpp/Grpc_h.hs

```diff
@@ -141,10 +144,68 @@ inline #{className}::#{proxyName}<TThreadPool>::#{proxyName}(
     { }
 
 #{doubleLineSep 0 methodDecl serviceMethods}
+
+#{template}struct #{className}::Schema
+{
+    #{export_attr}static const ::bond::Metadata metadata;
+
+    #{newlineSep 1 methodMetadata serviceMethods}
+
+    public: struct service
+    {
+        #{doubleLineSep 2 methodTemplate serviceMethods}
+    };
+
+    private: typedef boost::mpl::list<> methods0;
+    #{newlineSep 1 pushMethod indexedMethods}
+
+    public: typedef #{typename}methods#{length serviceMethods}::type methods;
+
+    #{constructor}
+};
+#{onlyTemplate $ CPP.schemaMetadata cpp s}
 |]
       where
         className = CPP.className s
         template = CPP.template s
+        onlyTemplate x = if null declParams then mempty else x
+        typename = onlyTemplate [lt|typename |]
+
+        export_attr = optional (\a -> [lt|#{a}
```

Comment:
We should update the DLL example to access metadata to make sure this works as we expect.

Reply:
Will do.